wri/wri-odp
# Add support for GFW dataset migrations

To test the migration:

- Make sure the following variables are set in the CKAN `.env`:
  ```
  CKANEXT__WRI__MIGRATION_DEPLOYMENT_NAME=migration_deployment
  CKANEXT__WRI__MIGRATION_FLOW_NAME=trigger-migration
  CKANEXT__WRI__MIGRATION_DEPLOYMENT_ENV=local # this should match the end of the Prefect variables/secret, e.g., `ckan-api-key-local`
  ```
- Start your local Docker environment
- Create the team `global-forest-watch` in your local CKAN (the file you will use has all datasets assigned to that team)
- `cd migration`
- Create Python virtual env and install requirement
- `export FLOW_DEPLOYMENT_ENV=local`
- In the Prefect UI, add these variables:
  - `ckan_frontend_url_local=http://localhost:3000`
  - `ckan_url_local=http://ckan-dev:5000`
- In the Prefect UI, add this Block (type Secret):
  - `ckan-api-key-local=A_LOCAL_SYS_ADMIN_API_KEY`
- Run the migration Prefect `main.py` (I like to output to a file):
  - `python main.py > gfw-migration.log 2>&1`
- Trigger the migration:
  ```
  curl -H "Authorization: API_KEY" "http://ckan-dev:5000/api/3/action/trigger_migration?file_name=gfw_datasets.csv"
  ```
